mm2src/mm2_gui_storage/src/account/mod.rs

```diff
@@ -0,0 +1,121 @@
+use crate::account::storage::AccountStorageError;
+use mm2_err_handle::prelude::*;
+use mm2_number::BigDecimal;
+use rpc::v1::types::H160 as H160Json;
+use serde::{Deserialize, Serialize};
+use serde_repr::{Deserialize_repr, Serialize_repr};
+use std::collections::BTreeSet;
+
+pub(crate) mod storage;
+
+pub const MAX_ACCOUNT_NAME_LENGTH: usize = 255;
+pub const MAX_ACCOUNT_DESCRIPTION_LENGTH: usize = 600;
+pub const MAX_TICKER_LENGTH: usize = 255;
+
+pub(crate) type HwPubkey = H160Json;
+
+#[derive(Clone, Copy, Debug, Deserialize_repr, Serialize_repr)]
+#[repr(u8)]
+pub(crate) enum AccountType {
+    Iguana = 0,
+    HD = 1,
+    HW = 2,
+}
+
+impl TryFrom<i64> for AccountType {
+    type Error = MmError<AccountStorageError>;
+
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(AccountType::Iguana),
+            1 => Ok(AccountType::HD),
+            2 => Ok(AccountType::HW),
+            other => {
+                let error = format!("Unknown 'account_type' value: {}", other);
+                MmError::err(AccountStorageError::ErrorDeserializing(error))
+            },
+        }
+    }
+}
+
+/// An enable account type.
+/// We should not allow the user to enable [`AccountType::HW`].
```

Comment:
Could you please provide more info on why we should not allow user to do it? It seems even a bit confusing to me because there is `device_pubkey` field in `gui_account` SQLite storage table, which I consider as HW device related (might be wrong, though 🙂 )

Reply:
`Enabled` account is an account that is currently enabled by the user. So after the restart GUI needs to activate the enabled account by default. As we discussed with @yurii-khi, in the current implementation, if the user wants to login with a Hardware wallet, he needs to plug-in his device, activate it etc.
So Hardware wallet can't be activated by default on restart.
However we need to store `device_pubkey` in `enabled_account_table` to declare a FOREIGN KEY, because the FOREIGN KEY needs to consist of the same columns as a PRIMARY KEY in `account_table`.